src/ucs/sys/sys.c

```diff
@@ -327,6 +330,16 @@ ucs_open_output_stream(const char *config_str, ucs_log_level_t err_log_level,
             return UCS_ERR_IO_ERROR;
         }
 
+        if (p_filename != NULL) {
+            *p_filename = ucs_strdup(filename, "filename");
```

Comment:
why need strdup? can't just return pointer inside the input string? since filename is last token anyway

Reply:
we can't do it, since can be generated name in case of `%t`/`%h`/`%p`